docs/proposals/sysctl.md

```diff
@@ -0,0 +1,477 @@
+
+Supported sysctls (whitelist):
+
+- network namespace: `net.*`
+
+Applied changes:
+
+* https://github.com/coreos/rkt/issues/2140
+
+Issues:
+
+* https://github.com/coreos/rkt/issues/2075
+
+## Design Alternatives and Considerations
+
+- Each pod has its own network stack that is shared amongs its containers.
+  A privileged side-kick or init container (compare https://github.com/kubernetes/contrib/blob/master/ingress/controllers/nginx/examples/sysctl/change-proc-values-rc.yaml#L80)
+  is able to set `net.*` sysctls.
+
+  Clearly, this is completely uncontrolled by the kubelet, but is a usable work-around if privileged
+  containers are permitted in the environment. As privileged container permissions (in the admission controller) are an all-or-nothing
+  decision and the actual code executed in them is not limited, allowing privileged container might be a security threat.
+
+  The same work-around als works for shared memory and message queue sysctls as they are shared among the containers of a pod
+  in their ipc namespace.
+
+- Instead of giving the user a way to set sysctls for his pods, an alternative seems to be to set high values
+  for the limits of interest from the beginning inside the kubelet or the runtime. Then - so the theory - the
+  user's pods operate under quasi unlimited bounds.
+
+  This might be true for some of the sysctls, which purely set limits for some host resources, but
+
+  * some sysctls influence the behavior of the application, e.g.:
+    * `kernel.shm_rmid_forced` adds a garbage collection semantics to shared memory segments when possessing processes die.
+      This is against the System V standard though.
+    * `net.ipv4.tcp_abort_on_overflow` makes the kernel send RST packets when the application is overloaded, giving a load-balancer
+      the chance to reschedule a request to another backend.
+  * some sysctls lead to changed resource requirement characteristics, e.g.:
+    * `net.ipv4.tcp_rmem`/`net.ipv4.tcp_wmem` not only define min and max values, but also the default tcp window buffer size
+      for each socket. While large values are necessary for certain environments and applications, they lead to waste of resources
+      in the 90% case.
+  * some sysctls have a different error behavior, e.g.:
+    * creating a shared memory segment will fail immediately when `kernel.shmmax` is too small.
+
+      With a large `kernel.shmmax` default, the creation of a segment always succeeds, but the OOM killer will
+      do its job when a shared memory segment exceeds the memory request of the container.
+
+  The high values that could be set by the kubelet on launch might depend on the node's capacity and capabilities. But for
+  portability of workloads it is helpful to have a common baseline of sysctls settings one can expect on every node. The
+  kernel defaults (which are active if the kubelet does not change defaults) are such a (natural) baseline.
+
+## Analysis of Sysctls on the initial Whitelist
```

Comment:
I believe we should explicitly call out the recommendation that folks should use taints and tolerations when landing pods that would like to modify these values for the 1st iteration b/c we simply do not know what the impact will be be.  The fences are light at best and non-existent for others. 

/cc @davidopp